### PR TITLE
PERF: read_sas

### PR DESCRIPTION
**Pull request by jbrockmendel**
```
from asv_bench.benchmarks.io.sas import *
self8 = SAS()
self8.setup("sas7bdat")

%timeit self8.time_read_sas("sas7bdat")
41.6 ms ± 1.77 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)  # <- master
22.7 ms ± 1.04 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)  # <- PR

```

### CODE REVIEW

**Review by jreback — APPROVED** [review]
lgtm. @bashtage if any comments, merge when ready

File: pandas/io/sas/sas7bdat.py [hunk 1]
```diff
@@ -798,25 +798,26 @@ def _chunk_to_dataframe(self) -> DataFrame:
...
-                rslt[name] = self._string_chunk[js, :]
+                rslt[name] = pd.Series(self._string_chunk[js, :], index=ix)
                 if self.convert_text and (self.encoding is not None):
                     rslt[name] = rslt[name].str.decode(
                         self.encoding or self.default_encoding
                     )
                 if self.blank_missing:
                     ii = rslt[name].str.len() == 0
-                    rslt.loc[ii, name] = np.nan
+                    rslt[name][ii] = np.nan
```
Comment (bashtage) [inline pandas/io/sas/sas7bdat.py:816]: Is chained assignment future proof?

Comment (jreback) [reply pandas/io/sas/sas7bdat.py:816]: rslt is a dict here :->

Comment (bashtage) [reply pandas/io/sas/sas7bdat.py:816]: Ahhh

**Pull request merged**